packages/dcos-ui-update-service/extra/dcos-ui-update-service.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=DC/OS UI Update Service: service for DC/OS Update
+[Service]
+Restart=always
+RestartSec=5
+PermissionsStartOnly=True
+User=dcos_ui_update_service
+SupplementaryGroups=dcos_adminrouter
+EnvironmentFile=/opt/mesosphere/environment
+EnvironmentFile=/opt/mesosphere/etc/dcos-ui-update-service.env
+EnvironmentFile=-/run/dcos/etc/dcos-ui-update-service.env
+ExecStartPre=/opt/mesosphere/bin/bootstrap dcos-ui-update-service
+ExecStart=/opt/mesosphere/bin/dcos-ui-update-service --verbose=true
+Sockets=dcos-ui-update-service.socket
```

Comment:
I'm not sure I would use systemd to manage your socket file unless you really need that. For starters I would remove the `Sockets=` directive and the `*.socket` file and just tell your program to listen on a given network (`unix`) and address (`/run/dcos/dcos-ui-update-service.sock`).

Reply:
Hmm - maybe sockets are actually better - they let you specify the file ownership, which is nice for restricting access to your service from Admin Router. In that case, you'll want to use https://github.com/coreos/go-systemd/tree/master/activation to obtain the listener that your server listens on.

Reply:
How to obtain it?

Reply:
Here's a better example than mesos-dns:
https://github.com/mesosphere/dcos-enterprise/tree/master/packages/dcos-backup/extra

It's pretty similar to what you have right now.

For a code sample of how this is used see:
https://github.com/mesosphere/dcos-backup/blob/master/main.go#L45-L63

Reply:
I have no access to the second repository, I will check the first though.

Reply:
Let's get you that access.

Reply:
Right - the socket code you have is perfect - you now need to adjust your Go program to be able to receive connections through systemd. That's what the link to dcos-backup is for, it serves as a solid example of how to do that. I asked to have you granted access to that repository, I'm not sure whether you have it yet. Please follow up on that.

Reply:
Please check https://github.com/dcos/dcos/pull/3566/commits/dda368074f37499991d12c0884177419751438e8 to see if I did it right. I still don't have access to that repository, I guess it might take some time.

Reply:
I'll ping J now.

Reply:
I have access now.